core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -952,6 +952,15 @@ private[spark] class SparkSubmit extends Logging {
     } catch {
       case t: Throwable =>
         throw findCause(t)
+    } finally {
+      if (!isShell(args.primaryResource) && !isSqlShell(args.mainClass) &&
```

Comment:
Can we check if we're in K8s here ?

Reply:
+1 to making it k8s specific ...
Ideally, we should be using daemon threads for most of our background work ...

Reply:
+1

Reply:
Since SPARK-34674 was released already via Apache Spark 3.1.2, I will make another JIRA for this, @HyukjinKwon , @mridulm , @Ngone51 .

Reply:
Here is the PR.
- https://github.com/apache/spark/pull/33403